test/Ocelot.Benchmarks/HeavyRoutesStartupBenchmark.cs

```diff
@@ -0,0 +1,162 @@
+            AddDiagnoser(MemoryDiagnoser.Default);
+            AddValidator(BaselineValidator.FailOnError);
+        }
+
+        [GlobalSetup]
+        public void SetUp()
+        {
+            //Setting up for more than 600 routes in different files
+            CreateOcelotConfigFile(0,100,"ocelot.json");
+            CreateOcelotConfigFile(101, 500, "ocelot.second.json");
+        }
+
+/* * Summary *
+        BenchmarkDotNet v0.13.11, Windows 11 (10.0.22621.2861/22H2/2022Update/SunValley2)
+  Intel Core i7-8650U CPU 1.90GHz(Kaby Lake R), 1 CPU, 8 logical and 4 physical cores
+  .NET SDK 8.0.100
+  [Host]     : .NET 6.0.25 (6.0.2523.51912), X64 RyuJIT AVX2[AttachedDebugger]
+  DefaultJob : .NET 6.0.25 (6.0.2523.51912), X64 RyuJIT AVX2
+| Method                             | Mean    | Error    | StdDev   | StdErr   | Min     | Q1      | Median  | Q3      | Max     | Op/s   | Gen0       | Gen1      | Allocated |
+|----------------------------------- |--------:|---------:|---------:|---------:|--------:|--------:|--------:|--------:|--------:|-------:|-----------:|----------:|----------:|
+| StartOcelotWithLargeConfigurations | 5.350 s | 0.0519 s | 0.0460 s | 0.0123 s | 5.214 s | 5.339 s | 5.351 s | 5.376 s | 5.408 s | 0.1869 | 42000.0000 | 3000.0000 | 185.05 MB |//        BenchmarkDotNet v0.13.11, Windows 11 (10.0.22621.2861/22H2/2022Update/SunValley2)
+  Intel Core i7-8650U CPU 1.90GHz(Kaby Lake R), 1 CPU, 8 logical and 4 physical cores
+  .NET SDK 8.0.100
+  [Host]     : .NET 6.0.25 (6.0.2523.51912), X64 RyuJIT AVX2[AttachedDebugger]
+  DefaultJob : .NET 6.0.25 (6.0.2523.51912), X64 RyuJIT AVX2
+| Method                             | Mean    | Error    | StdDev   | StdErr   | Min     | Q1      | Median  | Q3      | Max     | Op/s   | Gen0       | Gen1      | Allocated |
+|----------------------------------- |--------:|---------:|---------:|---------:|--------:|--------:|--------:|--------:|--------:|-------:|-----------:|----------:|----------:|
+| StartOcelotWithLargeConfigurations | 5.350 s | 0.0519 s | 0.0460 s | 0.0123 s | 5.214 s | 5.339 s | 5.351 s | 5.376 s | 5.408 s | 0.1869 | 42000.0000 | 3000.0000 | 185.05 MB |
+        */
+
+        [Benchmark]
+        public void StartOcelotWithLargeConfigurations()
+        {
+            OcelotStartup($"http://localhost:{FindAvailablePort()}");
+        }
+
+        [GlobalCleanup]
+        public void Cleanup()
+        {
+            // Stop the web host after benchmarking
+            _ocelot.StopAsync().GetAwaiter().GetResult();
+        }
+
+        private int FindAvailablePort()
+        {
+            TcpListener listener = null;
+            try
+            {
+                listener = new TcpListener(IPAddress.Loopback, 0);
+                listener.Start();
+                return ((IPEndPoint)listener.LocalEndpoint).Port;
+            }
+            finally
+            {
+                listener?.Stop();
+            }
+        }
```

Comment:
Pretty new logic?
Why not to use helper from `Ocelot.Testing` project. But the `PortFinder` uses `Socket` to bind to.
You use `TcpListener`... So it can be a new helper in `Ocelot.Testing` project

Reply:
done